### PR TITLE
Handle cancel state for ethereum txs

### LINKED ISSUE #1382: Do not remove cancelled txs from recent activities
# Summary

With the new tx speed up check (introduced on https://github.com/gnosis/cowswap/issues/1302) cancelled network transactions are removed from the UI once they are mined.

This change proposes to not remove them, and set the status `cancelled` instead.

image

---

For context:

> > @alfetopito I tried to reproduce that case consistently but couldn't. From what I see, the only way for a successful toast to be shown is for that tx to be mined (has a receipt), and if that is the case it's an unrelated issue to this PR.
> 
> Yes, could have happened as you said, maybe by the time that I sent my cancellation tx it has already been mined.
> I'll try again to be sure

Indeed, could not reproduce it.

I've tested the following flows, all starting with a tiny gas price:
1. Speed up. Detected and updated fine :heavy_check_mark: 
2. Failure (out of gas https://rinkeby.etherscan.io/tx/0x7621f4092bd04b766053d180400ca8ddc0b7d51e168f4d521a89b181019a40ab). :heavy_check_mark: 
3. Cancellation :white_check_mark: 

The only concern this time is that for cancelled txs, they were removed from the UI completely.

I guess that's out of the scope of this task, so I'll create a new one for it.

Besides that, good to go for me.

_Originally posted by @alfetopito in https://github.com/gnosis/cowswap/issues/1302#issuecomment-911927861_

### PR DESCRIPTION
**Pull request by ramirotw**
# Summary

Closes #1382 #1435

This adds a `Cancelling` and `Cancelled` states for ethereum transactions in the Recent Activity list. Before cancelled txs were removed from the list when they were confirmed.

The changes are built on top of @anxolin's PR https://github.com/gnosis/cowswap/pull/1429 as I'm making use of the new `EnhancedTransactionDetails` type and also to reduce conflicts due to changes in the `enhancedTransactions` state files.

image

  # To Test

1. Run an onchain transaction (wrap eth or approve) with very low gas fees
2. Cancel it through MM 
3. The transaction log in the Recent Activity list should change to `Cancelling` and then `Cancelled` when confirmed

### CODE REVIEW

**Comment by elena-zh** [comment]
Hi there!
I do not what I'm doing wrong, but I can't verify this issue, due to each time when I cancel a transaction, I get it stuck in 'Open' status. Each of these transaction navigates to a not found transaction log
image
Approve: 
https://rinkeby.etherscan.io/tx/0x86aa24b660c876ade79f1bdcca830b047ab323634747ebed3c958a8a15c348f9
Unwrap: https://rinkeby.etherscan.io/tx/0x1f05656a69f4ec33d0c2b6e4c51eac09398ae92a83316182b2bd1e00d3ba1cd9
Wrap: https://rinkeby.etherscan.io/tx/0x70875d4dd9f9a86d66d075a6705846b14202a5fc25f25455bb53c0fc561ab02a

Transaction logs in MM: 
image

File: src/custom/hooks/useRecentActivity.ts [hunk 1]
```diff
@@ -137,12 +137,12 @@ export function useActivityDescriptors({ chainId, id }: { chainId?: ChainId; id:
       // setup variables accordingly...
       const isReceiptConfirmed =
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/custom/hooks/useRecentActivity.ts:138]: That's on me. David asked the same thing. Answer: https://github.com/gnosis/cowswap/pull/1533#discussion_r725195634

**Comment by alfetopito** [comment]
> Hi there! I do not what I'm doing wrong, but I can't verify this issue, due to each time when I cancel a transaction, I get it stuck in 'Open' status. Each of these transaction navigates to a not found transaction log image Approve: https://rinkeby.etherscan.io/tx/0x86aa24b660c876ade79f1bdcca830b047ab323634747ebed3c958a8a15c348f9 Unwrap: https://rinkeby.etherscan.io/tx/0x1f05656a69f4ec33d0c2b6e4c51eac09398ae92a83316182b2bd1e00d3ba1cd9 Wrap: https://rinkeby.etherscan.io/tx/0x70875d4dd9f9a86d66d075a6705846b14202a5fc25f25455bb53c0fc561ab02a
> 
> Transaction logs in MM: image

My guess is that this is related to my changes to blocknative api key.
Needs to have this change merged in https://github.com/gnosis/cowswap/pull/1534 and a manual setting change on blocknative

I'll do that and let you know when you can test it

**Comment by ramirotw** [comment]
> But still I see 'Confirmed' status instead of 'Cancelled', and transaction details navigate to 404 page

@elena-zh please re test

**Comment by elena-zh** [comment]
Hey @ramirotw , thanks!
Now I'm able to see "Cancelling' and 'Cancelled' statuses for transactions. 

However, I have faced 2 issues during testing changes. 
1. I got a stuck 'Approve' transaction in 'Cancelling' state. View transaction details navigate to 404 error page: 
pending cancelling
Navigation from Cowswap: https://rinkeby.etherscan.io/tx/0x9bf9ee1ecb2d2c54666abde3d1a1ddd0cb6babe2ea570ed67adc176c1811884a
Navigation from the wallet: https://rinkeby.etherscan.io/tx/0x3fd3df9acdf0c1adcda09f66dc6146769c562ce74b2c8e359d49ae2966fb3ddb
 Transaction's history from the wallet
image

2. The case might not be related to the existing one, but I was able to reproduce it: I got failed transaction in stuck 'Pending' state. But before it failed, I tried to speed it up. The issue might be related to #1410 case, though...
failed as pending
Transaction details from the CowSwap: https://rinkeby.etherscan.io/tx/0x4f0f4497751252f4c03ae77fdeaf57126123e5a2ae0856ca25c61681906d042b
Transaction details from the wallet: https://rinkeby.etherscan.io/tx/0x4f0f4497751252f4c03ae77fdeaf57126123e5a2ae0856ca25c61681906d042b

**Comment by ramirotw** [comment]
> If I'm not mistaken, all described issues might be related to #1832 task.
> If yes, changes under the current PR LGTM. 🙂

You are right, this is probably related to the race condition issue already in https://github.com/gnosis/cowswap/issues/1832

**Pull request merged**